test/extensions/filters/http/wasm/config_test.cc

```diff
@@ -337,7 +341,8 @@ TEST_P(WasmFilterConfigTest, YamlLoadFromRemoteWasmFailCachedThenSucceed) {
   Init::ExpectableWatcherImpl init_watcher2;
 
   EXPECT_CALL(context_, initManager()).WillRepeatedly(ReturnRef(init_manager2));
-  factory.createFilterFactoryFromProto(proto_config, "stats", context_);
+  EXPECT_THROW_WITH_MESSAGE(factory.createFilterFactoryFromProto(proto_config, "stats", context_),
+                            WasmException, "Unable to create Wasm HTTP filter ");
```

Comment:
Yay for mocks. I cannot tell from this test whether production code will continue or crash here.

Reply:
Yah, you need to have an intimate knowledge of xDS and how filter configuration interacts with it.  Pour one out for clear interfaces and compartmentalized design.